for code review

JKuskys opened the pull request.

JKuskys committed:
added components: home-page, footer,registration,header

neja5785 committed:
Update package.json

neja5785 committed:
Update package.json

neja5785 committed:
Update package.json

neja5785 committed:
Update package.json

neja5785 committed:
Update package.json

neja5785 committed:
Update package.json

neja5785 committed:
Update package.json

JKuskys committed:
registration halfway done

neja5785 committed:
Create server.js

neja5785 committed:
Update package.json

neja5785 committed:
Update package.json

neja5785 committed:
Update package.json

neja5785 committed:
Update package.json

neja5785 committed:
Update package.json

neja5785 committed:
Update package.json

neja5785 committed:
heroku

neja5785 committed:
heroku

neja5785 committed:
Update package.json

neja5785 committed:
Update server.js

neja5785 committed:
Update server.js

neja5785 committed:
heroku

neja5785 committed:
heroku

neja5785 committed:
heroku

neja5785 committed:
heroku

JKuskys committed:
Merge branch 'master' of https://github.com/JKuskys/IT-academy-project-front-end

JKuskys committed:
registration modal

JKuskys committed:
registration sumbition, no proper validation

Cirtautaite committed:
started to create home page

JKuskys committed:
added material.angular.io

JKuskys committed:
created branch registration

JKuskys committed:
validation works

JKuskys committed:
Merge pull request #2 from JKuskys/registration

Registration

ModestaC committed:
Text design (#1)

Co-authored-by: Modesta <m.cirtautaite@gmail.com>

JKuskys committed:
registration post working (#3)

JKuskys committed:
heroku fix?

JKuskys committed:
Merge branch 'master' of https://github.com/JKuskys/IT-academy-project-front-end

JKuskys committed:
heroku fix 2

JKuskys committed:
heroku fix 3

ModestaC committed:
Text (#4)

* Text design

* Added image, aligned text and button design

Co-authored-by: Modesta <m.cirtautaite@gmail.com>
Co-authored-by: JKuskys <48916581+JKuskys@users.noreply.github.com>

JKuskys committed:
Heroku (#5)

* heroku 4)

* heroku 5

JKuskys committed:
Styles (#6)

* heroku 4)

* heroku 5

* styles update

neja5785 committed:
Welcomepage1 (#7)

* login

* login

* login

Co-authored-by: JKuskys <48916581+JKuskys@users.noreply.github.com>

JKuskys committed:
Registration approve (#8)

* #0.5 testing

* registration_messages implemented

JKuskys committed:
posts, but no api path (#9)

Cirtautaite committed:
Changed design and text

Cirtautaite committed:
Merge branch 'master' of https://github.com/JKuskys/IT-academy-project-front-end

JKuskys committed:
readjusted home-page (#10)

mildapetr committed:
adding responsive ui to home page (#11)

JKuskys committed:
Home page readjustment (#12)

* registration is reseting properly after close

* login clears when closed

neja5785 committed:
Registration login design (#15)

* login

Co-authored-by: jkuskys <j.kuskys1@gmail.com>
Co-authored-by: JKuskys <48916581+JKuskys@users.noreply.github.com>

JKuskys committed:
registraion bug fix+scrollbar

JKuskys committed:
post and error messaages working

mildapetr committed:
Merge pull request #16 from JKuskys/postModalaUpdateFixes

post and error messages working

neja5785 committed:
bugai (#17)

mildapetr committed:
feature/form-responsive-ui (#18)

* making header adjustments

* fixing home page issues

* making registration form responsive

mildapetr committed:
a very very small adjustment

JKuskys committed:
hot fixes for demo (#19)

JKuskys committed:
true/false post tiny fix

neja5785 committed:
fixed (#20)

mildapetr committed:
fix/responsive-ui (#21)

* fixed responsive issues with phone number

* fixed responsive issues successful registration popup

* fixed responsive issues login

neja5785 committed:
bugai (#23)

JKuskys committed:
It5.2 (#24)

* blocks are created

* routes added

* error fix

neja5785 committed:
Bugs (#28)

* bugai

* bugs

Co-authored-by: JKuskys <48916581+JKuskys@users.noreply.github.com>

matasgraj commented on prettier.config.json:
```diff
@@ -0,0 +1,10 @@
+module.exports = {
```
Duplicate file

JKuskys replied:
fixed

matasgraj commented on src/app/Services/modal/modal.service.ts:
```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({ providedIn: 'root' })
+export class ModalService {
```
Is this still needed? Material dialog does things like this over the hood

JKuskys replied:
It's removed now, won't be seen in later updates.

neja5785 committed:
Validation (#29)

* bugai

* bugs

* validation bug fixed

Co-authored-by: JKuskys <48916581+JKuskys@users.noreply.github.com>

matasgraj commented on src/app/Services/user.service.ts:
```diff
@@ -0,0 +1,46 @@
+})
+export class UserService {
+  private url = 'https://academy-project-back.herokuapp.com/';
+  private readonly apiPath = '/api';
```
unused

JKuskys replied:
seems fixed

JKuskys committed:
gets correct response from server when user logs int (#25)

* gets correct response from server if the user is either logged in or not, but registration is not working after last backend update

* registration post body fix

matasgraj commented on src/app/app.module.ts:
```diff
@@ -0,0 +1,146 @@
+import {MatTreeModule} from '@angular/material/tree';
+import { ApplicationBlockComponent } from './application-block/application-block.component';
+
+const appRoutes: Routes = [
```
If you are using routes here, delete 'app-routing.module'

JKuskys replied:
moved routes to app-routing.module

matasgraj commented on src/app/app.module.ts:
```diff
@@ -0,0 +1,146 @@
+    CdkTableModule,
+    CdkTreeModule,
+    DragDropModule,
+    MatAutocompleteModule,
```
Do you really need all these Mat components?

JKuskys replied:
Commented most of them for now, just so it would be easy to fix, if something stoped working, will fully delete unused ones, when everything is tested

matasgraj commented on src/app/application-block/application-block.component.html:
```diff
@@ -0,0 +1,23 @@
+      </div>
+    </div>
+    <div class="col-sm align-self-center ">
+      <div *ngIf="application.status === 'accepted'" class=" p-1 status-passed">Patvirtinta</div>
```
You can use *ngSwitch here

matasgraj commented on src/app/application-block/application-block.component.scss:
```diff
@@ -0,0 +1,45 @@
+  padding-left: 6px
+}
+.status-null{
+  background-color: rgba(105, 105, 105, 0.55);
```
From my experience hex codes are used more. But it's your choosing, my request is that all code would have one type

matasgraj commented on src/app/application-block/application-block.component.scss:
```diff
@@ -0,0 +1,45 @@
+.not-seen-style{
+  padding-left: 6px
+}
+.status-null{
```
These 4 classes are almost the same. Create 'base' class with border-radius, width, and text-align, and then you could change only background-color with other css classes

JKuskys replied:
whole component will be unused and deleted

matasgraj commented on src/app/header/header.component.scss:
```diff
@@ -0,0 +1,62 @@
+    text-align: justify;
+  }
+
+  .container-text2{
```
well that doesn't say anything, update naming

matasgraj commented on src/app/header/header.component.ts:
```diff
@@ -0,0 +1,26 @@
+    this.dialog.open(LoginComponent);
+  }
+ 
+  openModal(id: string) {
```
is this still needed?

JKuskys replied:
fixed

matasgraj commented on src/app/home-page/home-page.component.ts:
```diff
@@ -0,0 +1,57 @@
+  applications: Application[];
+
+  constructor(private modalService: ModalService, private dialog: MatDialog) {
+    this.applications = [{
```
why it's in home page comp?

JKuskys replied:
if you are asking about application array, then it was for testing purpose, it has been deleted

matasgraj commented on src/app/home-page/home-page.component.ts:
```diff
@@ -0,0 +1,57 @@
+  }
+
+  ngOnInit(): void {
+    this.bodyText = 'This text can be updated in modal 1';
```
.

matasgraj commented on src/app/registration/registration.component.html:
```diff
@@ -0,0 +1,310 @@
+                </div>
+            </div>
+
+            <!-- <mat-form-field class="example-full-width">
```
delete unused code

JKuskys replied:
fixed

matasgraj commented on src/app/registration/registration.component.scss:
```diff
@@ -0,0 +1,215 @@
+    border-radius: 25px;
+}
+html,
+body {
```
hm, why this is in registration component? I guess it should be in main styles.scss

matasgraj commented on src/app/registration/registration.component.ts:
```diff
@@ -0,0 +1,183 @@
+    private httpService: HttpClient,
+    private userService: UserService,
+    private modalService: ModalService,
+    private dialog: MatDialogRef<any>,
```
so this should be named as `dialogRef` and `dialogNew` renamed to `dialog`

matasgraj commented on src/app/registration/registration.component.ts:
```diff
@@ -0,0 +1,183 @@
+
+  constructor(
+    private fb: FormBuilder,
+    private httpService: HttpClient,
```
do you really need httpclient?

JKuskys replied:
moved it to services

matasgraj commented on src/app/registration/registration.component.ts:
```diff
@@ -0,0 +1,183 @@
+  }
+
+  ngOnInit(): void {
+    this.httpService.get('./assets/phone-codes.json').subscribe(
```
this should be in its own service

JKuskys replied:
moved it to services

matasgraj commented on src/app/registration/registration.component.ts:
```diff
@@ -0,0 +1,183 @@
+    this.passwordNotMatch = false;
+    this.submission = false;
+    this.info = {
+      // id: Math.floor(Math.random() * 10),
```
? dont you need an id?

JKuskys replied:
It's autoincremented in backend

ModestaC committed:
added applications component (#30)

Co-authored-by: Modesta <m.cirtautaite@gmail.com>

matasgraj commented on src/app/shared/application.ts:
```diff
@@ -0,0 +1,18 @@
+export interface Application {
+  id?: number;
+  name: string;
+  phone_number?: string;
```
camelCase. underscore is used in uppercase constants.

JKuskys replied:
Refactored it, it will be in camelCase in later updates

matasgraj commented on src/app/shared/registration.ts:
```diff
@@ -0,0 +1,17 @@
+import {User} from './user';
+
+export interface Info {
```
isn't its the same as `Application` interface? At least it shares most of the parameters. https://www.typescriptlang.org/docs/handbook/interfaces.html#indexable-types

JKuskys replied:
they are a bit different now, not sure what would be the best practice

matasgraj replied:
use extend. `export interface Dog extends Animal {...}`. More info in link i pasted in this thread

matasgraj reviewed (changes requested):
use prettier to format your code and delete old modal functionality

JKuskys committed:
Features/it3 (#34)

* gets correct response from server if the user is either logged in or not, but registration is not working after last backend update

* registration post body fix

* login works, ready for testing

* error fix

* err fix

* still errors

JKuskys committed:
Fix/application (#35)

* gets correct response from server if the user is either logged in or not, but registration is not working after last backend update

* registration post body fix

* login works, ready for testing

* error fix

* err fix

* still errors

* fixed issues with application list showing

matasgraj commented on src/app/Services/auth-service.service.ts:
```diff
@@ -0,0 +1,18 @@
+  constructor() {
+  }
+
+  // ...
```
don't leave comments that don't say/explain anything

JKuskys replied:
fixed

mildapetr committed:
feature/IT5.3 (#37)

* adding simple application details view

* starting on admin application view (WIP)

* minimal style adjustments

* adding comments to detailed view with mock data

* restyling application details page based on ux feedback

* improved information flow for application

* adding admin comment writing placeholder

* not really changing anything, need another pull request

* fixing conflicts

JKuskys committed:
fix(while not logged in user gets redirected properly) (#38)

mildapetr committed:
fixing CORS issues and getting application data from backend (#39)

neja5785 committed:
Applicationlist (#40)

* aplikaciju sarasas

* application list

JKuskys committed:
Fix/login (#41)

* fix(while not logged in user gets redirected properly)

* paraiškų mygtukas pridėtas

JKuskys committed:
able to get from server

JKuskys committed:
Fix/bugs (#42)

* fixed bugs(password validation, fromWhere field validation, hide registration button, when logged in, buttons according to roles, only when false on contract isselected comment tab appears

* registration button is not covered anymore

neja5785 committed:
loading-spinners (#44)

Co-authored-by: JKuskys <48916581+JKuskys@users.noreply.github.com>

JKuskys committed:
Fix/bugs (#43)

* fixed bugs(password validation, fromWhere field validation, hide registration button, when logged in, buttons according to roles, only when false on contract isselected comment tab appears

* registration button is not covered anymore

* fixed registration password validation

neja5785 committed:
Error regex (#45)

* loading-spinners

* regex for password

JKuskys committed:
properly loading everywhere (#46)

* properly loading everywhere

* fixed compilation error

mildapetr committed:
feature/IT5.6 (#47)

* adding application comment loading (comments not for specific application though, waiting for endpoint)

* adding application comment loading (comments not for specific application though, waiting for endpoint)

* some progress with comments

* added status change from detailed view

JKuskys committed:
fixed password validation (#48)

mildapetr committed:
added status change when visiting detailed view (#49)

mildapetr committed:
fixed naming difference for 'seen' status (#50)

JKuskys committed:
Fix/password validation (#51)

* fixed password validation

* cleaning up the code

* role fix(proper things done according to role)

* import of mat fix

JKuskys committed:
imports fix (#52)

JKuskys committed:
user application created, no comments (#53)

neja5785 committed:
Bugs and fixes (#54)

* loading-spinners

* regex for password

* table sort

* a few fixes

* marginas headerio buttonam

* hovers

mildapetr committed:
fix/application-comments (#55)

* fixed blank comment accepting

* added comment editability

* improved comment data flow and added delete

* adjusting comment system to back-end changes

* adding status change snack bar

* fixed blank comment accepting

added comment editability

improved comment data flow and added delete

adjusting comment system to back-end changes

adding status change snack bar

* fixing compiling issues

JKuskys committed:
new application design (#56)

matasgraj commented on src/app/Services/application/comment.service.ts:
```diff
@@ -0,0 +1,33 @@
+
+  constructor(private httpClient: HttpClient) { }
+
+  getComments({ applicationId }): Observable<Comment[]> {
```
why an object?

matasgraj commented on src/app/Services/authorization/role-guard-service.service.ts:
```diff
@@ -16,14 +18,28 @@ export class RoleGuardService implements CanActivate {
 
+  setRole(token: string) {
+    const tokenPayload = this.jwtHelper.decodeToken(token);
+    if (isNotNullOrUndefined(tokenPayload)) {
```
just go with a simple 'if' statement, don't use `isNotNullOrUndefined`. It will go 'true' if the value is not null or undefined.

JKuskys replied:
fixed

matasgraj commented on src/app/Services/universal/custom-validators.ts:
```diff
@@ -0,0 +1,18 @@
+import {AbstractControl, ValidationErrors, ValidatorFn} from '@angular/forms';
+
+export class CustomValidators {
```
isn't this the same file as `.../Services/custom.validators.ts`  ? Also 'Services' folder should be in lower-case.

JKuskys replied:
fixed

matasgraj commented on src/app/admin-application-details/admin-application-details.component.html:
```diff
@@ -0,0 +1,28 @@
+<div id="admin-application-details-wrapper">
+  <mat-progress-spinner
+    style="margin:0 auto;"
```
no inline style in html, move to scss

JKuskys replied:
fixed

matasgraj commented on src/app/admin-application-details/admin-application-details.component.ts:
```diff
@@ -0,0 +1,88 @@
+export class AdminApplicationDetailsComponent implements OnInit {
+  isLoading = false;
+  private routeSub: Subscription;
+  public application: Application;
```
it's not necessary to add `public` modifier because by default it's `public`

JKuskys replied:
fixed

matasgraj commented on src/app/admin-application-details/admin-application-details.component.ts:
```diff
@@ -0,0 +1,88 @@
+
+  ngOnInit(): void {
+    this.isLoading = true;
+    this.routeSub = this.route.params.subscribe(params => {
```
well.. you have subscribe in a subscribe in a subscribe. It's bad coding practice with the possibility to have major memory leaks. Let's say it's okay to have the first subscribe block for route params and for fetching application. So after getting an application, call another function with needed params that will call backend for comment (line 34). In that case, the outer subscribe should destroy itself.

mildapetr replied:
fixed (used fork join instead to run the requests in parallel)

matasgraj commented on src/app/admin-application-details/admin-application-details.component.ts:
```diff
@@ -0,0 +1,88 @@
+          this.application = data;
+          this.comments = commentData;
+          this.isLoading = false;
+          if (this.application.status === 'NAUJA') {
```
this block also could be moved to its own function. `if (this.application.status == 'NAUJA') { this.updateCommentStatus(commentID)}`

JKuskys replied:
Fixed

matasgraj commented on src/app/admin-application-details/admin-application-details.component.ts:
```diff
@@ -0,0 +1,88 @@
+          this.application = data;
+          this.comments = commentData;
+          this.isLoading = false;
+          if (this.application.status === 'NAUJA') {
```
also, create an enum in a new file for those values: 'NAUJA', 'PERZIURETA' etc. and update all files where these values are used with enums

JKuskys replied:
Fixed

matasgraj commented on src/app/admin-application-details/admin-application-details.component.ts:
```diff
@@ -0,0 +1,88 @@
+
+  onCommentSaved(input: string): void {
+    let newComment: Comment;
+    newComment = {
```
this could be done in one line less `const newComment: Comment = { ...}`

JKuskys replied:
fixed

matasgraj commented on src/app/admin-application-details/admin-application-details.component.ts:
```diff
@@ -0,0 +1,88 @@
+      commentDate: formatDate(new Date(), 'yyyy-MM-dd', 'en'),
+      comment: input
+    };
+    this.commentService.addComment(newComment, {applicationId: this.application.id}).subscribe(res => {
```
`...subscribe(res: Comment => this.comments.push(res));`

JKuskys replied:
fixed

matasgraj commented on src/app/admin-application-details/admin-application-details.component.ts:
```diff
@@ -0,0 +1,88 @@
+  }
+
+  onCommentEdited(comment: Comment): void {
+    this.commentService.updateComment(comment, { applicationId: this.application.id }).subscribe();
```
do you really need to subscribe here when you don't care about response?

mildapetr replied:
well, it doesn't seem to actually send the request to the back-end unless subscribe() is present. Not sure how to fix this

matasgraj commented on src/app/admin-comment/admin-comment.component.ts:
```diff
@@ -0,0 +1,44 @@
+  switchEditMode(): void {
+    this.inEditMode = !this.inEditMode;
+    if (this.inEditMode) {
+      this.commentForm.get('commentBody').setValue(this.comment.comment);
```
you should try to avoid these kinds of namings `this.comment.comment`

matasgraj commented on src/app/student-comment/student-comment.component.scss:
```diff
@@ -0,0 +1,16 @@
+.student-comment{
+  //background:  #f0f0f0;
```
remove

JKuskys replied:
removed

JKuskys committed:
application redesign (#57)

JKuskys committed:
Features/it6.1 (#58)

* application redesign

* post of email and getting data is working

mildapetr committed:
fix/comment-section (#59)

* changing edit and delete to lithuanian names

* changing comment formating

* adding special remark if no comments are written

JKuskys committed:
Fix/it2 (#60)

* bug fix for IT2

* code cleap-up

mildapetr committed:
adding responsive table design for application list table (#61)

mildapetr committed:
fixed application fetching subscription issues (#62)

JKuskys committed:
sessions are properly working (#63)

mildapetr committed:
feature/IT7.3 (#64)

* adding feedback comment writing and reading in admin detailed view

* adding validation for comments

neja5785 committed:
Student comments (#65)

* getting conversation

* conversation

* conversation

JKuskys committed:
Feature/it8.2 (#67)

* feature IT8.2 implemented

* fix for not loading comments

JKuskys committed:
Bug/it5 (#68)

* feature IT8.2 implemented

* fix for not loading comments

* registration post properly and date in applications comments are shown normally

matasgraj commented on src/app/application-details/application-details.component.ts:
```diff
@@ -17,10 +18,42 @@ export class ApplicationDetailsComponent implements OnInit {
+
+  potential(): boolean {
+    if (this.application.status === Status.POTENCIALUS &&
+      localStorage.getItem('roles').includes('ADMIN')) {
```
roles should be used with enum

matasgraj commented on src/app/registration/registration.component.ts:
```diff
@@ -135,28 +139,40 @@ export class RegistrationComponent implements OnInit {
         Validators.required,
-        Validators.maxLength(1500)
+        Validators.maxLength(1500),
+        // check if there are only gaps
```
i guess 1 comment is more than enough :)

matasgraj commented on src/app/shared/status.ts:
```diff
@@ -0,0 +1,7 @@
+export enum Status {
+  NAUJA = 'NAUJA',
```
https://www.typescriptlang.org/docs/handbook/enums.html#string-enums 

enum property should be in sentence case. `status { Nauja = 'NAUJA', ...}`

mildapetr committed:
fix/responsive-application-view (#70)

* fixing status font size to be responsive

* fixing applicant name to be responsive

mildapetr committed:
adding hamburger menu for mobile view (#69)

mildapetr committed:
fixed buttons disappearing sometimes (#71)

mildapetr committed:
fixed student comment writing still possible even though admin comment was not written first (#72)

JKuskys committed:
when token expires proper logout happens (#73)

ModestaC committed:
editing tests so that they pass (#74)

Co-authored-by: Modesta <m.cirtautaite@gmail.com>

JKuskys committed:
Unit test working (#75)

mildapetr committed:
fix/comment-loading (#76)

* adding loading spinner for admin comments

* adding loading spinner for student comments

JKuskys committed:
Unit test (#77)

* Unit test working

* unit tests added

mildapetr committed:
minor style and layout changes (#78)

JKuskys committed:
feature IT10 implemented (#79)

JKuskys committed:
feature IT17 implemented (#80)

mildapetr committed:
feature/IT11.1 (#81)

* adding file upload

* adding comment attachment when writing a comment

* adding comment download from server

* styling attachment download section

JKuskys committed:
clickable (#82)

JKuskys committed:
Feature/it13.1 (#83)

* feature from admin side implemented

* code refactor

* cleaned up the code

mildapetr committed:
feature/IT11.3 (#85)

* improving comment attachment styling

* adding validation for attachment

neja5785 committed:
Clarification (#84)

* clarification

* sticky

* clarification and modal

* fixeS

mildapetr committed:
making application take .docx file as well (#86)

ModestaC committed:
Limit comments per page, tabs (#87)

Co-authored-by: Modesta <m.cirtautaite@gmail.com>
Co-authored-by: JKuskys <48916581+JKuskys@users.noreply.github.com>

neja5785 committed:
Clarification (#88)

* clarification

* sticky

* clarification and modal

* fixeS

mildapetr committed:
fix/IT16 (#89)

* fixing merge issues and introducing other way of loading more comments for student

* fixing merge issues and improving admin application view comment display

neja5785 committed:
Clarification (#90)

* clarification

* sticky

* clarification and modal

* fixeS

* word delete

JKuskys committed:
Features/it4 (#91)

* added reset password page

* Route updatePassword added

JKuskys committed:
Fix/notification (#92)

* added reset password page

* Route updatePassword added

* notification fix

matasgraj commented on src/app/header/header.component.spec.ts:
```diff
@@ -1,6 +1,19 @@
+import { RouterTestingModule } from '@angular/router/testing';
+import {RoleGuardService} from '../services/authorization/role-guard-service.service';
+
+class MockAuthServiceService extends AuthServiceService{
```
you can create variable with a service type, like `let mockAuthService: AuthService`

mildapetr committed:
adding some minimal styling changes (#93)

JKuskys committed:
functionality works, but not fully (#94)

neja5785 committed:
styles (#95)

Co-authored-by: neja5785 <nedas.jankauskas@mif.stud.vu.lt>

matasgraj commented on src/app/forgot-password/forgot-password.component.ts:
```diff
@@ -0,0 +1,62 @@
+  templateUrl: './forgot-password.component.html',
+  styleUrls: ['./forgot-password.component.scss']
+})
+export class ForgotPasswordComponent implements OnInit {
```
PasswordReminderComponent

matasgraj commented on src/app/login/login.component.ts:
```diff
@@ -51,13 +58,13 @@ export class LoginComponent implements OnInit {
         this.serverErrorMessage = '';
         this.closeDialog('login');
-        if (localStorage.getItem('roles').includes('ADMIN')){
+        if (localStorage.getItem('roles').includes('ADMIN')) {
```
Use already created enum for roles.

matasgraj commented on src/app/password-reset/password-reset.component.html:
```diff
@@ -0,0 +1,87 @@
+          <input
+            matInput
+            type="password"
+            id="password1"
```
weird id

matasgraj commented on src/app/password-reset/password-reset.component.html:
```diff
@@ -0,0 +1,87 @@
+          <input
+            matInput
+            type="password"
+            id="password2"
```
weird id

matasgraj reviewed (changes requested):
Rename services with duplicate 'service' word. "somthing-service.service.ts", should be "something.service.ts"

mildapetr committed:
fixing some home page issues (#96)

JKuskys committed:
added messages and proper routing (#97)

mildapetr committed:
fixing some issues with save status button (#98)

mildapetr committed:
fixing very stupid mistake (#99)

JKuskys committed:
validation of fullname fix (#100)

mildapetr committed:
fixing application list button layout (#101)

JKuskys committed:
new unitTests (#102)
